packages/report-server/src/reportBuilder/transform/parser/functions/context.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { reduceToDictionary } from '@tupaia/utils';
+
+import { Context, ContextProp } from '../../../context';
+
+/**
+ * `contextProps`: props that will be required by the function
+ * `create`: factory method that creates the function by injecting the context
+ */
+type ContextParams = {
+  contextProps: ContextProp[];
+  create: (deps: { getContext: () => Context }) => (...args: any[]) => unknown;
+};
```

Comment:
I used `params` since it is used in a seemingly similar vein throughout the project: get an object (eg from `transform`/`fetch` config), and construct a function (the `transform`/`fetch` function). I guess that my bias towards "shorter names where possible" played a role too!

In retrospect, those are not quite the same thing, so happy to change that - it feels like equating oranges and apples here 😄 

About `contextDependencies`: that was the term I used initially. I also tried to inject the `context` object directly (instead of a getter for it: `getContext`), but math.js wouldn't let me. I think I quite like `contextProps` eventually, because that's what this is: a prop of the context - the fact that `context` is eventually injected as a "dependency", through a getter, is probably an implementation detail. It also reminds me a bit of `react` props which I feel is an advantage - you props are drilled down your components, and you don't care where they came from...

Happy to discuss this further though!